src/config/rollup.config.js

```diff
@@ -88,8 +88,19 @@ if (isPreact) {
 }
 
 const externalPattern = new RegExp(`^(${external.join('|')})($|/)`)
-const externalPredicate =
-  external.length === 0 ? () => false : id => externalPattern.test(id)
+
+function externalPredicate(id) {
+  const isDep = externalPattern.test(id)
+  if (umd) {
+    // for UMD, we want to bundle all non-peer deps
+    return isDep
+  }
+  // for esm/cjs we want to make all node_modules external
+  // TODO: support bundledDependencies if someone needs it ever...
+  const isNodeModule = id.includes('node_modules')
```

Comment:
i'd make it more strict with RegExp `/\/node_modules\//`, u'd have to check if `id` is normalized here, if not u'd have to construct the regexp based on path.sep (to support windows)